agent/pkg/k8s/k8s.go

```diff
@@ -92,23 +92,24 @@ func NewClient(sharedInformers informers.SharedInformerFactory, k8sClient *l5dk8
 	// For types that we do not have a cleint for we use the `unstructured` package. As we
 	// also do not have proto definitions for these CRDs, we serialize them to JSON.
 	//
-	// +-------------------------+-----------------------------+----------+------------+
-	// |          Group          |            Kind             |  Client  | Serializer |
-	// +-------------------------+-----------------------------+----------+------------+
-	// | policy.linkerd.io       | servers,serverAuthorizaions | dynamic  | json       |
-	// | multicluster.linkerd.io | links                       | dynamic  | json       |
-	// | linkerd.io              | serviceprofiles             | spclient | json       |
-	// | split.smi-spec.io       | trafficsplits               | tsclient | json       |
-	// +-------------------------+-----------------------------+----------+------------+
+	// +-------------------------+---------------------+-----------+------------+
+	// |          Group          |        Kind         |  Client   | Serializer |
+	// +-------------------------+---------------------+-----------+------------+
+	// | policy.linkerd.io       | serverAuthorizaions | l5dClient | json       |
+	// | policy.linkerd.io       | servers             | l5dClient | json       |
+	// | multicluster.linkerd.io | links               | dynamic   | json       |
+	// | linkerd.io              | serviceprofiles     | spclient  | json       |
```

Comment:
i noticed we only reference `spClient` in `main.go` when we do:
```go
spClient, err := spclient.NewForConfig(k8sConfig)
```
... and then `spClient` is passed into `k8s.NewClient()`, which takes that param as `l5dClient l5dApi.Interface`. is there another client we should instead be instantiating? feels weird to create an `spClient` and then cast it to a more generic `l5dClient`?

Reply:
Right.. This is me not renaming things properly... So `spclient` is imported as: 

```
spclient "github.com/linkerd/linkerd2/controller/gen/client/clientset/versioned"
```

This is the case because before this just contained the ServiceProfile client. Now it contains the server authorization and server clients as well so it really should be: 


```
l5dApi "github.com/linkerd/linkerd2/controller/gen/client/clientset/versioned"
```

as this is not just the SP client anymore.  I have changed this. The reason we are taking an `l5dApi.Interface` instead of a `l5dApi.ClientSet` is to enable passing a fake client there, which we do in tests.